ee/desktop/menu/menu_template.go

```diff
@@ -0,0 +1,89 @@
+package menu
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/kolide/kit/version"
+)
+
+type TemplateDataOption func(*templateData)
+
+func WithVersion(info version.Info) TemplateDataOption {
+	return func(td *templateData) {
+		td.LauncherVersion = info.Version
+		td.LauncherRevision = info.Revision
+		td.GoVersion = info.GoVersion
+	}
+}
+
+func WithOSQueryVersion(osqueryVersion string) TemplateDataOption {
+	return func(td *templateData) {
+		td.OSQueryVersion = osqueryVersion
+	}
+}
+
+func WithHostname(hostname string) TemplateDataOption {
+	return func(td *templateData) {
+		td.Hostname = hostname
+	}
+}
+
+func WithLogFilePath(logFilePath string) TemplateDataOption {
+	return func(td *templateData) {
+		td.LogFilePath = logFilePath
+	}
+}
+
+func WithLauncherFlagsFilePath(launcherFlagsFilePath string) TemplateDataOption {
+	return func(td *templateData) {
+		td.LauncherFlagsFilePath = launcherFlagsFilePath
+	}
+}
+
+type templateData struct {
+	LauncherVersion       string
+	LauncherRevision      string
+	GoVersion             string
+	OSQueryVersion        string
+	Hostname              string
+	LogFilePath           string
+	LauncherFlagsFilePath string
+}
+
+func NewTemplateData(opts ...TemplateDataOption) *templateData {
+	template := &templateData{
+		LauncherVersion:       "unknown",
+		LauncherRevision:      "unknown",
+		OSQueryVersion:        "unknown",
```

Comment:
```suggestion
		OsqueryVersion:        "unknown",
```

`osquery` is one word.